Source/HTTP/WinHttp/winhttp_http_task.cpp

```diff
@@ -1719,22 +1765,26 @@ void winhttp_http_task::callback_websocket_status_headers_available(
     {
         DWORD dwError = GetLastError();
         HC_TRACE_ERROR(HTTPCLIENT, "HCHttpCallPerform [ID %llu] [TID %ul] WinHttpWebSocketCompleteUpgrade errorcode %d", TO_ULL(HCHttpCallGetId(pRequestContext->m_call)), GetCurrentThreadId(), dwError);
+        pRequestContext->m_lock.unlock();
         pRequestContext->complete_task(E_FAIL, HRESULT_FROM_WIN32(dwError));
+        return;
```

Comment:
Can someone verify that this is the intended behavior?

Reply:
if pRequestContext->m_hRequest == NULL then you can't do the code below if that's the question...

---
In reply to: [632151443](https://github.com/microsoft/libHttpClient/pull/580#discussion_r632151443) [](ancestors = 632151443)